### PR TITLE
Python workflow with Dremio service, flak8 and pytest

### PR DESCRIPTION
**Pull request by tifflhl**
- Added workflow to start up Dremio service.
- In the same workflow, added steps to lint example.py.
- Also added e2e tests for example.py.
- example.py is formatted with standard indentation and block comments.

### CODE REVIEW

**Comment by tifflhl** [comment]
Please see https://github.com/tifflhl/arrow-flight-client-examples/actions for successful test runs.

File: .github/workflows/python-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+jobs:
+  test-with-dremio:
+    runs-on: ubuntu-latest
+    defaults:
+        run:
+          working-directory: python
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.5, 3.6, 3.7, 3.8]
```
Comment (kylep-dremio) [inline .github/workflows/python-workflow.yaml:26]: Why limited to these 4? Can we instead put in a lower bound only?

File: .github/workflows/python-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    services:
+      dremio:
+        image: dremio/dremio-oss:latest
+        ports:
+          - 9047:9047
```
Comment (kylep-dremio) [inline .github/workflows/python-workflow.yaml:32]: Please map only the ports in use to avoid this becoming out of date.

File: .github/workflows/python-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      - name: Lint with flake8
+        run: |
+          # Stop the build if there are Python syntax errors or undefined names
+          flake8 example.py --count --select=E9,F63,F7,F82 --show-source --statistics
+          # Exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 example.py --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+      # Below step can be improved to run a Python or Bash script that checks for Dremio's liveness
+      - name: Sleep before test to ensure that Dremio is started
+        run: |
+          sleep 180
```
Comment (kylep-dremio) [inline .github/workflows/python-workflow.yaml:59]: Please ensure there's a ticket to track improving this.

File: python/README.md [hunk 4]
```diff
@@ -13,7 +15,7 @@ This lightweight Python client application connects to the Dremio Arrow Flight s
   - `python3 example.py -host '<DREMIO_HOST>' -user '<DREMIO_USERNAME>' -pass '<DREMIO_PASSWORD>'`
 
 ```
-usage: example.py [-h] [-host HOSTNAME] [-port FLIGHTPORT] -user USERNAME -pass PASSWORD [-query SQLQUERY] [-tls] [-certs TRUSTEDCERTIFICATES]
+usage: example.py [-h] [-host HOSTNAME] [-port FLIGHTPORT] -user USERNAME -pass PASSWORD [-query SQLQUERY] [-tls] [-certs TRUSTEDCERTIFICATES] [-sn SERVERNAME]
```
Comment (kylep-dremio) [inline python/README.md:18]: Please pull SERVERNAME out into a separate PR.

File: python/test.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      pytest.xfail("Bad hostname.")
+      connect_to_dremio_flight_server_endpoint("badHostNamE",
+        "32010", "dremio", "dremio123", False, False, False, False)
+
+    def test_bad_port(self):
+      pytest.xfail("Bad port.")
+      connect_to_dremio_flight_server_endpoint("localhost",
+        "12345", "dremio", "dremio123", False, False, False, False)
+
+    def test_non_existent_user(self):
```
Comment (kylep-dremio) [inline python/test.py:42]: Invalid password?

**Review by kylep-dremio — CHANGES_REQUESTED** [review]

File: python/test.py (hunk 5 above)
Comment (tifflhl) [reply python/test.py]: Added test in new the new change.

File: python/README.md (hunk 4 above)
Comment (tifflhl) [reply python/README.md]: Removed SERVERNAME feature from this PR.

File: .github/workflows/python-workflow.yaml (hunk 3 above)
Comment (tifflhl) [reply .github/workflows/python-workflow.yaml:72]: Will keep track of it in Jira.

File: .github/workflows/python-workflow.yaml (hunk 2 above)
Comment (tifflhl) [reply .github/workflows/python-workflow.yaml]: Done.

File: .github/workflows/python-workflow.yaml (hunk 1 above)
Comment (tifflhl) [reply .github/workflows/python-workflow.yaml:42]: 3.9 is not compatible with PyArrow, please see comment in the updated PR.

**Comment by tifflhl** [comment]
Successful run in my fork https://github.com/tifflhl/arrow-flight-client-examples/actions/runs/330307227

**Comment by tifflhl** [comment]
Successful run on my fork https://github.com/tifflhl/arrow-flight-client-examples/actions/runs/330338990.

**Review by kylep-dremio — APPROVED** [review]
